### PR TITLE
Update location model

### LINKED ISSUE #59: Add location tests
We need to have tests for the location model.

### PR DESCRIPTION
**Pull request by Ofekmn**
adds verification and the ability to update.

resolves #59

### CODE REVIEW

**Review by talko611 — COMMENTED** [review]
LGTM, please add update method tests

**Comment by ormergi** [comment]
Hi Ofek, please take a look at my comment from another PR about tests that don't test the app business logic https://github.com/redhat-beyond/FitMeet/pull/26#discussion_r1171429021

**Review by talko611 — APPROVED** [review]

File: location/models.py [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
+        self.street = street
+
+    def update_street_number(self, street_number):
+        self.street_number = street_number
+
+    def update_indoor(self, indoor):
+        self.indoor = indoor
+
+    def update_description(self, description):
+        self.description = description
```
Comment (kobihk) [inline location/models.py:41]: maybe you can do one update method with all the options like:
```suggestion
    def update(self, name=None, city=None, street=None, street_number=None, indoor=None, description=None):
        if name:
            self.name = name
        if city:
            self.city = city
        if street:
            self.street = street
        if street_number:
            self.street_number = street_number
        if indoor is not None:
            self.indoor = indoor
        if description:
            self.description = description
```

Comment (kobihk) [reply location/models.py:41]: now in the test file, you can do all the tests in one test.
also, I think you can do something like "test_update_of_invalid_values"

**Review by kobihk — CHANGES_REQUESTED** [review]

File: location/models.py [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
     indoor = models.BooleanField()
     description = models.CharField(max_length=255)
 
     def __str__(self):
         return self.name
+
+    def update(self, name=None, city=None, street=None, street_number=None, indoor=None, description=None):
+        temp_location = Location.objects.get(id=self.id)
+        if name is not None:
+            temp_location.name = name
```
Comment (kobihk) [inline location/models.py:22]: I think it should be the same as using self
```suggestion
        if name is not None:
            self.name = name
            ...
```

**Review by rotem2022 — DISMISSED** [review]
LGTM

File: location/tests.py [hunk 3]
```diff
@@ -1,3 +1,95 @@
...
+LOCATION_NAME_NEW = "location2"
+CITY_NEW = 'city2'
+STREET_NEW = 'street2'
+STREET_NUMBER_NEW = 2
+INDOOR_NEW = True
+DESCRIPTION_NEW = 'description2'
+
+LONG_FIELD = "a" * 45
+EMPTY_FIELD = ''
+SPECIAL_CHARACTERS = "special%@chars"
```
Comment (kobihk) [inline location/tests.py:22]: you can also get all the special characters by doing:

```suggestion
import string

SPECIAL_CHARACTERS = string.punctuation
```
the output of string.punctuation is: '!"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~'

File: location/tests.py [hunk 4]
```diff
@@ -1,3 +1,95 @@
...
+        with pytest.raises(IntegrityError):
+            Location.objects.create(name=LOCATION_NAME, city=CITY_NEW, street=STREET_NEW,
+                                    street_number=STREET_NUMBER_NEW, indoor=INDOOR_NEW,
+                                    description=DESCRIPTION_NEW)
+
+    @pytest.mark.parametrize("name, city, street", [
+        (EMPTY_FIELD, CITY, STREET),  # empty name
+        (LOCATION_NAME, LONG_FIELD, STREET),  # long city name
+        (LOCATION_NAME, CITY, SPECIAL_CHARACTERS),  # street name with special chars
+    ],
```
Comment (kobihk) [inline location/tests.py:47]: you can enlarge the invalid tests to all the rest of the fields:
street_number, indoor, description

File: location/tests.py [hunk 5]
```diff
@@ -1,3 +1,95 @@
...
+        assert updated_location.street == STREET_NEW
+        assert updated_location.street_number == STREET_NUMBER_NEW
+        assert updated_location.indoor == INDOOR_NEW
+        assert updated_location.description == DESCRIPTION_NEW
+
+    @pytest.mark.parametrize("name, city, street", [
+        (EMPTY_FIELD, CITY, STREET),  # empty name
+        (LOCATION_NAME, LONG_FIELD, STREET),  # long city name
+        (LOCATION_NAME, CITY, SPECIAL_CHARACTERS),  # street name with special chars
+    ],
```
Comment (kobihk) [inline location/tests.py:88]: same here ^^
you can enlarge the invalid update tests to all the rest of the fields:
street_number, indoor, description

**Review by kobihk — CHANGES_REQUESTED** [review]

**Review by kobihk — APPROVED** [review]

**Review by rotem2022 — APPROVED** [review]

**Review by Sendy459 — APPROVED** [review]

**Review by ygalblum — APPROVED** [review]

**Pull request merged**